### PR TITLE
do not require incoming rtp metadata

### PR DESCRIPTION
**Pull request by mat-hek**

### CODE REVIEW

File: lib/rtp_h264/payloader.ex [hunk 1]
```diff
@@ -129,126 +128,94 @@ defmodule Membrane.RTP.H264.Payloader do
...
-  defp delete_prefix(@frame_prefix_shorter <> rest), do: rest
-
-  defp clear_parser_acc(state) do
-    state
-    |> Map.put(:parser_acc, [])
-    |> Map.put(:acc_byte_size, 0)
-    |> Map.put(:stap_a_reserved, 0)
-    |> Map.put(:stap_a_nri, 0)
-  end
+  defp timestampify(buffers, %{timestamp_resolution: resolution}) do
```
Comment (bblaszkow06) [inline lib/rtp_h264/payloader.ex:211]: This name is awful :stuck_out_tongue:, use sth like `add_rtp_timestamps` or `calc_rtp_timestamps`

File: lib/rtp_h264/payloader.ex [hunk 2]
```diff
@@ -129,126 +128,94 @@ defmodule Membrane.RTP.H264.Payloader do
...
+    alias Membrane.Time
 
-  defp update_stap_a_properties(<<r::1, nri::2, _type::5, _rest::binary()>> = payload, state) do
-    state
-    |> Map.put(:acc_byte_size, state.acc_byte_size + byte_size(payload))
-    |> Map.put(:stap_a_reserved, state.stap_a_reserved && r)
-    |> Map.put(:stap_a_nri, min(state.stap_a_nri, nri))
+    Enum.map(buffers, fn %Buffer{metadata: metadata} = buffer ->
+      rtp_timestamp = Time.to_seconds(resolution * metadata.timestamp)
+      metadata = Map.put(metadata, :rtp, %{timestamp: rtp_timestamp})
```
Comment (bblaszkow06) [inline lib/rtp_h264/payloader.ex:217]: That's not really rtp timestamp as it should start from random value. This is rather offset

**Review by bblaszkow06 — CHANGES_REQUESTED** [review]

File: lib/rtp_h264/plugin_app.ex [hunk 3]
```diff
@@ -0,0 +1,17 @@
+defmodule Membrane.RTP.H264.Plugin.App do
+  @moduledoc false
+  use Application
+  alias Membrane.RTP.{H264, PayloadFormat}
+
+  def start(_type, _args) do
+    PayloadFormat.register(%PayloadFormat{
```
Comment (bblaszkow06) [inline lib/rtp_h264/plugin_app.ex:7]: I think I haven't seen that in core, but looks cool! :smile:

File: lib/rtp_h264/payloader.ex [hunk 4]
```diff
@@ -102,8 +96,7 @@ defmodule Membrane.RTP.H264.Payloader do
     type = get_unit_type(payload, state)
     {acc_buffers, state} = handle_accumulator(type, buffer, state)
     {new_buffers, state} = handle_unit_type(type, buffer, state)
-    buffers = timestampify(acc_buffers ++ new_buffers, state)
-    {{:ok, buffer: {:output, buffers}, redemand: :output}, state}
+    {{:ok, buffer: {:output, acc_buffers ++ new_buffers}, redemand: :output}, state}
```
Comment (bblaszkow06) [inline lib/rtp_h264/payloader.ex:99]: What will be the timestamp of outgoing packet if it doesn't have rtp timestamp on entry (e.g. comes from encoder)?

File: lib/rtp_h264/payloader.ex [hunk 5]
```diff
@@ -89,7 +83,7 @@ defmodule Membrane.RTP.H264.Payloader do
 
   @impl true
   def handle_prepared_to_playing(_ctx, state) do
-    {{:ok, caps: {:output, %RTP{payload_type: state.payload_type}}}, state}
+    {{:ok, caps: {:output, %RTP{}}}, state}
```
Comment (bblaszkow06) [inline lib/rtp_h264/payloader.ex:86]: How will be the payload type provided?

Comment (bblaszkow06) [reply lib/rtp_h264/payloader.ex:86]: BTW tests seem to miss that part

**Review by bblaszkow06 — CHANGES_REQUESTED** [review]

File: lib/rtp_h264/payloader.ex (hunk 4 above)
Comment (mat-hek) [reply lib/rtp_h264/payloader.ex:99]: it's a job of the serializer now

File: lib/rtp_h264/payloader.ex (hunk 5 above)
Comment (mat-hek) [reply lib/rtp_h264/payloader.ex:86]: It's also moved to the serializer

File: lib/rtp_h264/plugin_app.ex (hunk 3 above)
Comment (mat-hek) [reply lib/rtp_h264/plugin_app.ex:7]: This is a part of membrane_rtp_format

**Pull request merged**